libpermian/plugins/koji/__init__.py

```diff
@@ -88,18 +95,21 @@ def to_compose(self):
         wait_until = None if timeout <= 0 else datetime.datetime.now() + datetime.timedelta(seconds=timeout)
         entrypoint = f'{self.composes_baseurl}/{self.task_id}-{self.package_name}'
         entrypoint_dir = os.path.dirname(entrypoint)
-        # try to locate the compose until timeout is reached
+
+        LOGGER.debug(f'Trying to locate the compose {entrypoint} until timeout {wait_until}')
         while wait_until is None or datetime.datetime.now() < wait_until:
-            response = requests.get(entrypoint)
-            if response.ok:
-                compose_relpaths = response.text.strip()
-                compose_relpath = compose_relpaths.split('\n')[-1]
-                compose_path = f'{entrypoint_dir}/{compose_relpath}'
-                try:
-                    compose_id = productmd.compose.Compose(compose_path).info.compose.id
-                    return ComposeStructure(self.settings, compose_id, location=compose_path)
-                except RuntimeError: # raised by productmd when failed to load compose metadata
-                    pass
+            try:
+                with urllib.request.urlopen(entrypoint, timeout=5) as response:
+                    compose_relpaths = response.read().decode().strip()
+                    compose_relpath = compose_relpaths.split('\n')[-1]
+                    compose_path = f'{entrypoint_dir}/{compose_relpath}'
+                    try:
+                        compose_id = productmd.compose.Compose(compose_path).info.compose.id
+                        return ComposeStructure(self.settings, compose_id, location=compose_path)
```

Comment:
It would be great to add LOGGER.debug printing compose path and compose id before returning it.

Reply:
Otherwise, it looks good.

Reply:
Good idea, done